src/os/windows/comptr.h

```diff
@@ -0,0 +1,91 @@
+/* $Id$ */
+
+/*
+ * This file is part of OpenTTD.
+ * OpenTTD is free software; you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, version 2.
+ * OpenTTD is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details. You should have received a copy of the GNU General Public License along with OpenTTD. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+ /** @file comptr.h COM object wrapper. */
+
+#ifndef COMPTR_H
```

Comment:
FWIW, src/sound/xaudio2_s.cpp makes use of #include <wrl/client.h> and the Microsoft::WRL::ComPtr class. This may have a minimum Windows SDK version, but given we use it elsewhere in OpenTTD, it doesn't seem unreasonable to me to use it here too.

Reply:
Oh, from all MS variants of COM smart pointer I didn't see that one. I think mine was a bit nicer to use, but I replaced it now with WRL one.